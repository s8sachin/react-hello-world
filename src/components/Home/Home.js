import React, { Component } from 'react';
import { connect } from 'react-redux';
import { submitValueAction } from '../../actions/myAction';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = { inputVal: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({ inputVal: e.target.value });
  }

  handleSubmit() {
    const { inputVal } = this.state;
    this.props.submitValueAction(inputVal);
  }

  render() {
    const { inputVal } = this.state;
    const { inputValueFromReducer } = this.props;
    return (
      <div>
        <input value={inputVal} onChange={this.handleChange} /><br />
        <button type="button" onClick={this.handleSubmit}>Submit</button> <br />
        <span>Your value : {inputValueFromReducer}</span>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { inputValueFromReducer } = state.myReducer;
  return { inputValueFromReducer };
};

export default connect(mapStateToProps, { submitValueAction })(Home);
