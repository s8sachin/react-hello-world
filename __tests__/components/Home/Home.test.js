import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReduxThunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import Home from '../../../src/components/Home';

configure({ adapter: new Adapter() });

const middlewares = [ReduxThunk];
const mockStore = configureStore(middlewares);

describe('Home', () => {
  it('should load without crashing', () => {
    const store = mockStore({ myReducer: { inputValueFromReducer: '' } });
    const wrapper = mount(<Home store={store} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
