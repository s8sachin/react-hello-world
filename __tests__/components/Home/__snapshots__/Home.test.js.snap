// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should load without crashing 1`] = `
<Connect(Home)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Home
    inputValueFromReducer=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    submitValueAction={[Function]}
  >
    <div>
      <input
        onChange={[Function]}
        value=""
      />
      <br />
      <button
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
       
      <br />
      <span>
        Your value : 
      </span>
    </div>
  </Home>
</Connect(Home)>
`;
